accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/configreader/YamlMacroElEvaluator.java

```diff
@@ -72,8 +72,26 @@ public FunctionMapper getFunctionMapper() {
             public VariableMapper getVariableMapper() {
                 return variableMapper;
             }
-        };
 
+            @Override
+            public Object convertToType(Object obj, Class<?> type) {
+                if(obj == null) {
+                    return null;
+                }
+                if(type == null) {
+                    return obj;
+                }
+                if(type.equals(String.class)) {
+                    return String.valueOf(obj);
+                }
+                if(type.isAssignableFrom(obj.getClass())) {
+                    return obj;
+                } else {
+                    // no special conversions supported
+                    throw new IllegalStateException("Cannot convert "+obj.getClass() +" to " + type + " (object: "+obj+")");
+                }
+            }
+        };

```

Comment:
Explicitly converting here (and we pretty much don't need any conversions) fixes the problem. Otherwise it would also have been possible to set the context class loader, but this seems cleaner to me @kwin

Reply:
LGTM, any chance we can catch such issues with a test?

Reply:
the TEST did not fail because it's one simple classloader (instead of OSGi) - not sure if writing a test for it is worth the effort (and it's not easy to do)